### PR TITLE
Topic recognition s4764094: GCN Model on Facebook Dataset

### PR DESCRIPTION
**Pull request by ChenyiHU-233**
In this task,  GCN neural network is used to predict and classify the partially processed Facebook large page-page network dataset and achieved an relatively high accuracy at about 70.76%. This neural network model uses node features and neighboring information to classify the nodes. For Facebook dataset, it has four distinct class targets, and each sample has 128 features which are used for the classification. This project aims to solve the multi-class node classification problem with reasonable accuracy.

This project includes the files as follows:
* dataset.py loads and preprocess the large facebook dataset, spilting it into 0.7 of train and 0.3 of test sets.
* modules.py stores the structure of our GCN model.
* predict.py has the function of tsne to visualize the performance.
* train.py main project is running here, used for model training and evaluation process.
* README.md contains the report of this project.

Thank you!
s4764094
Chenyi Hu

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**
GNN FB Page-Page Network dataset ---- Normal Difficulty
| Category	| | Marks | Comments |	
| :---  | :---: | :---: | :---  |		
| Algorithm solves the problem 		| 5 | 4.5 | Some overfitting seen which is unusual as you also have early stop? |
| Implementation functions as intended	| 3 | 2.5 | 1 layer GCN, torch_geometric/GCNConv. Model seems a bit too shallow, perhaps 2 layers would have been better. |
| Good design 					| 1 | 1 | Modular, Reusable |
| Commenting 					| 1 | 1 | Meaningful docstrings and comments |
| Algorithm above Normal Difficulty 	| 5 | 5 | |
| Algorithm is Hard difficulty 			| 5 | 0 | Normal Difficulty |
| Section IV : Max mark 15 from 20		|   | **14**   | |

**TSNE / UMAP:** 
Separation seems to have some issues. 
Needs more generalization for semantic separation, most probably due to low acc from only having 1 GNN layer.  
- Discussion: Good, with a bit more depth useful.

**Suggestions:**
Model would benefit from 2 or 3 layers.
Could add a hyper parameter search or discuss how the values were obtained.
GCN code is **not reusable ** as values are hard coded.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Some overfitting seen. Needs more GCN layers.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. GNN Task.</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. Usage info missing.</td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Good. 11</td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. Add some info about the specific implementation here.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-11</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
